subsys/net/lib/socks/socks.c

```diff
@@ -10,65 +12,79 @@ LOG_MODULE_REGISTER(net_socks, CONFIG_SOCKS_LOG_LEVEL);
 #include <zephyr.h>
 #include <net/socket.h>
 #include <net/socks.h>
+#include <net/net_pkt.h>
 
 #include "socks_internal.h"
 
-static int socks5_tcp_send(int fd, u8_t *data, u32_t len)
+static void socks5_method_rsp_cb(struct net_context *ctx,
+				 struct net_pkt *pkt,
+				 union net_ip_header *ip_hdr,
+				 union net_proto_header *proto_hdr,
+				 int status,
+				 void *user_data)
 {
-	u32_t offset = 0U;
-	int ret;
+	struct socks5_method_response *mthd_rep =
+			(struct socks5_method_response *)user_data;
 
-	while (offset < len) {
-		ret = send(fd, data + offset, len - offset, 0);
-		if (ret < 0) {
-			return ret;
-		}
+	if (!pkt || status) {
+		memset(mthd_rep, 0, sizeof(struct socks5_method_response));
+		goto end;
+	}
 
-		offset += ret;
+	if (net_pkt_read(pkt, (u8_t *)mthd_rep,
+			 sizeof(struct socks5_method_response))) {
+		errno = ENOBUFS;
```

Comment:
Hmm, will there even be a chance to utilize `errno` value assigned here? It's going to be overwritten in the `zsock_connect_ctx` function.